packages/components/nodes/retrievers/MeilisearchRetriever/MeilisearchRetriever.ts

```diff
@@ -0,0 +1,177 @@
+                host: host,
+                apiKey: meilisearchAdminApiKey
+            })
+            const flattenDocs = docs && docs.length ? flatten(docs) : []
+            const finalDocs = []
+            for (let i = 0; i < flattenDocs.length; i += 1) {
+                if (flattenDocs[i] && flattenDocs[i].pageContent) {
+                    const uniqueId = uuidv4()
+                    const { pageContent, metadata } = flattenDocs[i]
+                    const docEmbedding = await embeddings.embedQuery(pageContent)
+                    embeddingDimension = docEmbedding.length
+                    const documentForIndexing = {
+                        pageContent,
+                        metadata,
+                        objectID: uniqueId,
+                        _vectors: {
+                            ollama: {
+                                embeddings: docEmbedding,
+                                regenerate: false
+                            }
+                        }
+                    }
+                    finalDocs.push(documentForIndexing)
+                }
+            }
+            let index: any
+            try {
+                index = await client.getIndex(indexUid)
+            } catch (error) {
+                console.error('Error fetching index:', error)
+                await client.createIndex(indexUid, { primaryKey: 'objectID' })
+            } finally {
+                index = await client.getIndex(indexUid)
+            }
+
+            try {
+                await index.updateSettings({
+                    embedders: {
+                        ollama: {
+                            source: 'userProvided',
+                            dimensions: embeddingDimension
+                        }
+                    }
+                })
+                await index.addDocuments(finalDocs)
+            } catch (error) {
+                console.error('Error occurred while adding documents:', error)
+            }
+            return
+        }
+    }
+    async init(nodeData: INodeData, input: string, options: ICommonObject): Promise<any> {
+        const credentialData = await getCredentialData(nodeData.credential ?? '', options)
+        const meilisearchSearchApiKey = getCredentialParam('searchApiKey', credentialData, nodeData)
+        const host = nodeData.inputs?.host as string
+        const indexUid = nodeData.inputs?.indexUid as string
+        const K = nodeData.inputs?.K as string
+        const semanticRatio = nodeData.inputs?.semanticRatio as string
+        const embeddings = nodeData.inputs?.embeddings as Embeddings
+        const output = nodeData.outputs?.output as string
+
+        const hybridsearchretriever = new MeilisearchRetriever(host, meilisearchSearchApiKey, indexUid, K, semanticRatio, embeddings)
+        if (output == 'meilisearch retriever') {
+            return hybridsearchretriever
+        } else {
+            return hybridsearchretriever //TODO
```

Comment:
is there any TODO item that we still need to do here?

Reply:
yes , also like postgres, the component is intended to either return the documents retrieved with data type document , or you can return a baseRetriever to be used in the chain , but this is yet to be implemented, we can remove the extra output and only return the retriever for now and implement the other function in a future PR